src/pip/_internal/utils/misc.py

```diff
@@ -112,7 +112,7 @@ def get_pip_version():
 
     return (
         'pip {} from {} (python {})'.format(
-            __version__, pip_pkg_dir, sys.version[:3],
+            __version__, pip_pkg_dir, '{}.{}'.format(*sys.version_info),
```

Comment:
I think it would be helpful to add this as a helper function to `misc.py` since it's used so frequently and was a source of error, and the pattern isn't obvious to those who haven't seen it. It can be called something like `get_major_minor_version()`. It would also make it easier to track down all the places we're using this version string.

Reply:
might be a little premature DRY-ing :man_shrugging: 

excluding tests here's the usage count:

- `'python{}.{}'.format(*sys.version_info)`: 1
- `'...' % (sys.version_info[0], ...)`: 1
- `'{}.{}'.format(*sys.version_info)`: 3

Reply:
Yes, I know. I think it would be helpful for the reasons I stated above.

Reply:
shrugs, I don't think it's worth it -- if you feel strongly push the change to my branch

Reply:
As a maintainer, I think it's worth it. I will push this later today.

Reply:
looks like I missed a couple where the helper could be used:

- src/pip/_internal/utils/misc.py
- src/pip/_internal/locations.py